[receiver/solacereceiver] Solace Receiver (#10572)

mcardy opened the pull request:
**Description:** Add Solace receiver to receive trace data from a [Solace PubSub+ Event Broker](https://solace.com/products/event-broker/)

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/10572

**Testing:** Full test coverage using mocks

**Documentation:** Component documented in receiver's README.md

Sponsor: @djaglowski

mcardy committed:
Added Solace Receiver

mcardy committed:
Added entry to codeowners for Solace receiver

mcardy committed:
Merge branch 'main' of github.com:open-telemetry/opentelemetry-collector-contrib into open-telemetry-main

mcardy committed:
Merge pull request #1 from open-telemetry/main

Merge upstream

mcardy committed:
Fixed linting errors

Use uber's atomic package, use errors.Is instead of equality

mcardy committed:
Bumped dependency versions

mcardy committed:
Added solacereceiver to dependabot workflow

mcardy committed:
Added solacereceiver to CHANGELOG.md

mcardy committed:
Ran make gotidy

mcardy committed:
Refactored pdata package into model package

djaglowski commented:
@mcardy, please sign the CLA

djaglowski commented on receiver/solacereceiver/config.go:
```diff
@@ -0,0 +1,90 @@
+}
+
+type Transport struct {
+	TLS configtls.TLSClientSetting `mapstructure:"tls,omitempty"`
```
Is it necessary to nest this under `transport`? The vast majority of components using this config struct present `tls` at the top level of the config.

mgevantmakher replied:
we will present tls at the top level of the config

djaglowski reviewed (commented):
Overall, the code looks very clean and complete. I left just one comment. 

I'll trigger CI now, but the CLA check will be a blocker either way.

mcardy committed:
Restructured config moving transport.tls.* to tls.*

mcardy commented:
I've changed `transport.tls.*` to `tls.*`, thanks for the review. We're in the process of assigning the CLA Manager internally for this and will have the CLA signed soon.

mcardy committed:
Merge branch 'main' of github.com:open-telemetry/opentelemetry-collector-contrib into open-telemetry-main

mcardy committed:
Merge branch 'open-telemetry-main'

mcardy committed:
Updated Solace receiver to v0.55.0 of collector libraries

mcardy committed:
Make gotidy with proper version of collector dependencies

mcardy committed:
Updated protobuf spec to include service version, remove AD enqueue

A few other fields are also added and tests adjusted accordingly

Addresses SOL-73532, SOL-73533, SOL-73534

mcardy commented:
@djaglowski could you please run the checks for the latest commit and review the latest changes? We have no more expected changes for the time being.

djaglowski commented on receiver/solacereceiver/receiver.go:
```diff
@@ -0,0 +1,247 @@
+func (s *solaceTracesReceiver) Start(_ context.Context, _ component.Host) error {
+	recordReceiverStatus(receiverStateStarting)
+	var cancelableContext context.Context
+	cancelableContext, s.cancel = context.WithCancel(context.Background())
```
Great, thanks for explaining to me

mcardy committed:
Followed new changelog process to add a new component entry
